dissect/target/plugins/os/unix/linux/modules.py

```diff
@@ -0,0 +1,71 @@
+from dataclasses import dataclass
+from typing import Iterator
+
+from dissect.target.helpers.record import TargetRecordDescriptor
+from dissect.target.plugin import Plugin, export
+from dissect.target.target import Target
+
+ModuleRecord = TargetRecordDescriptor(
+    "unix/module",
+    [
+        ("string", "name"),
+        ("varint", "size"),
+        ("varint", "refcount"),
+        ("string[]", "used_by"),
+        ("path", "source"),
+    ],
+)
+
+
+@dataclass
+class Module:
+    path: str
+    name: str
+    size: int
+    refcnt: int
+    used_by: list[str]
+
+
+class ModulePlugin(Plugin):
+    def __init__(self, target: Target):
+        super().__init__(target)
+        self._module_paths = list(self.target.fs.path("/sys/module").iterdir())
```

Comment:
```suggestion
        self._module_paths = self.target.fs.path("/sys/module").iterdir()
```

I'd not resolve the full iterator here, just keep it as is.
The `check_compatible()` could then become:

```python
    def check_compatible(self) -> None:
        try:
            next(self._module_paths) :
        except StopIteration:
            raise UnsupportedPluginError("No module paths found.")
```

Reply:
Wouldn't this advance the generator, and you'd miss the first entry in the plugin itself? You'd also only be able to run the plugin once.

Reply:
Hmm, yes it would. Then just define the base path in `__init__` and `iterdir()` separately in each function.

Reply:
Alright I tried to follow the conversation. I think I fixed it by adding a base path in `__init__` and `next(base_path.iterdir())` in check_compatible. In the plugin I iterated again over the base path with `iterdir()`